### PR TITLE
Harden an ID conversion helper fn to help avoid a NPE

### PR DESCRIPTION
**Pull request by gbuisson**
Title is self explaining

connected to https://github.com/threatgrid/iroh/issues/2810

### CODE REVIEW

File: src/ctim/domain/id.cljc [hunk 1]
```diff
@@ -131,12 +131,15 @@
...
-(defn short-id->long-id [short-id url-params-fn]
-  (when-let [type (nth (re-matches short-id-re short-id) 2)]
-    (long-id
-     (short-id->id type
-                   short-id
-                   (url-params-fn)))))
+(s/defn short-id->long-id :- (s/maybe s/Str)
+  [short-id :- (s/maybe s/Str)
+   url-params-fn :- (s/maybe (s/=> {s/Keyword s/Any}))]
+  (when short-id
```
Comment (yogsototh) [inline src/ctim/domain/id.cljc]: Wouldn't it be better to log when `short-id` is `nil` because it might be a bug?

Comment (msprunck) [reply src/ctim/domain/id.cljc]: IMO it should not be logged in CTIM, it's like calling `map` with a `nil` coll. It should just return `nil`

File: src/ctim/domain/id.cljc [hunk 2]
```diff
@@ -131,12 +131,15 @@
...
+    (when-let [type (nth (re-matches short-id-re short-id) 2)]
```
Comment (ereteog) [inline src/ctim/domain/id.cljc]: you could remove first `when` and use [`not-found`parameter of `nth`](https://clojuredocs.org/clojure.core/nth):
`nth throws an exception unless not-found is supplied.`
```clojure
(when-let [type (nth (re-matches short-id-re short-id) 2 nil)]
...
)
```

**Comment by gbuisson** [comment]
I'm with Matt, I think logging should be done at levels above

**Pull request merged**